src/routers/alpha-router/functions/get-candidate-pools.ts

```diff
@@ -899,11 +903,25 @@ export async function getV2CandidatePools({
   // Filtering step for second hops
   const topByTVLUsingTokenInSecondHopsMap: Map<string, SubcategorySelectionPools<V2SubgraphPool>> = new Map();
   const topByTVLUsingTokenOutSecondHopsMap: Map<string, SubcategorySelectionPools<V2SubgraphPool>> = new Map();
-  const tokenInSecondHopAddresses = topByTVLUsingTokenIn.map((pool) =>
-    tokenInAddress == pool.token0.id ? pool.token1.id : pool.token0.id
+  const tokenInSecondHopAddresses = topByTVLUsingTokenIn.filter((pool) => {
+    // filtering second hops
+    if (tokenInAddress === pool.token0.id) {
+      return !tokensToAvoidOnSecondHops?.includes(pool.token1.id.toLowerCase());
+    } else {
+      return !tokensToAvoidOnSecondHops?.includes(pool.token0.id.toLowerCase());
+    }
+  }).map((pool) =>
+    tokenInAddress === pool.token0.id ? pool.token1.id : pool.token0.id
   );
-  const tokenOutSecondHopAddresses = topByTVLUsingTokenOut.map((pool) =>
-    tokenOutAddress == pool.token0.id ? pool.token1.id : pool.token0.id
+  const tokenOutSecondHopAddresses = topByTVLUsingTokenOut.filter((pool) => {
+    // filtering second hops
+    if (tokenInAddress === pool.token0.id) {
+      return !tokensToAvoidOnSecondHops?.includes(pool.token1.id.toLowerCase());
+    } else {
```

Comment:
do you need else if here:

```
if (tokenInAddress === pool.token0.id) {
   return !tokensToAvoidOnSecondHops?.includes(pool.token1.id.toLowerCase());
} else if (tokenInAddress === pool.token1.id) {
   return !tokensToAvoidOnSecondHops?.includes(pool.token0.id.toLowerCase());
}
```

Reply:
I think it is safe to assume that tokenInAddress is token1 in that case.